tests/discovery/test_discovery.py

```
@@ -40,19 +76,41 @@
+        ext = os.path.splitext(path)[1]
+        return HTTPD_EXTENSION_MAP.get(ext, super().guess_type(path))
+
+
+@pytest.fixture(scope="module", autouse=True)
+def http_server():
+    # Change working directory to the test files directory
+    os.chdir(TEST_FILES_DIR)
+
+    # Set up and start the HTTP server
+    handler = CustomHTTPRequestHandler
+    httpd = HTTPServer(("0.0.0.0", 8000), handler)
```

Ruff

Possible binding to all interfaces

(S104)